cdap-master/src/main/java/io/cdap/cdap/master/upgrade/PostUpgradeJobMain.java

```diff
@@ -65,19 +67,26 @@ public static void main(String[] args) {
     RetryStrategy retryStrategy =
       RetryStrategies.timeLimit(30, TimeUnit.SECONDS,
                                 RetryStrategies.exponentialDelay(10, 500, TimeUnit.MILLISECONDS));
+    final boolean restartSystemApps;
+    if (args.length == 4 && args[3].equals("true")) {
+      restartSystemApps = true;
```

Comment:
can be simplified to `restartSystemApps = args.length == 4 && "true".equals(args[3])`